src/KingSystem/Sound/sndMusicInfo.h

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <agl/Utils/aglParameter.h>
+#include <agl/Utils/aglParameterIO.h>
+#include <agl/Utils/aglParameterList.h>
+#include <agl/Utils/aglParameterObj.h>
+#include <container/seadBuffer.h>
+#include <resource/seadResource.h>
+#include "KingSystem/Utils/Types.h"
+
+namespace ksys::snd {
+
+class MusicInfo : public agl::utl::IParameterIO {
+public:
+    template <typename T>
+    struct MinMax : public agl::utl::IParameterObj {
+        inline MinMax(T min, T max, const sead::SafeString& name, const sead::SafeString& label,
+                      const sead::SafeString& meta, agl::utl::IParameterList* parent)
+            : mLabel(label), mMeta(meta), mMin(min, "Min", "最小", mMeta, this),
+              mMax(max, "Max", "最大", mMeta, this) {
+            parent->addObj(this, name);
+        }
+
+        sead::FixedSafeString<64> mLabel;
+        sead::FixedSafeString<32> mMeta;
+        agl::utl::Parameter<T> mMin;
+        agl::utl::Parameter<T> mMax;
+    };
+
+    MusicInfo();
+
+protected:
+    sead::SafeString field_1D0;
+    agl::utl::ParameterList mHeader;
+};
```

Comment:
size assertion?

Reply:
I haven't found any information about the size of the base class in code, so I can't fill the size in safely... it would just be a meaningless statement, as the size is taken from my definition instead of the game.

Reply:
Fair.